packages/studio-base/src/players/RosbridgePlayer.ts

```diff
@@ -546,85 +666,29 @@ export default class RosbridgePlayer implements Player {
   }
 
   public setPublishers(publishers: AdvertiseOptions[]): void {
-    // Since `setPublishers` is rarely called, we can get away with just throwing away the old
-    // Roslib.Topic objects and creating new ones.
-    for (const publisher of this.#topicPublishers.values()) {
-      publisher.unadvertise();
-    }
-    this.#topicPublishers.clear();
-    this.#advertisements = publishers;
-    this.#setupPublishers();
+	  publishers;
+	  this.#getServiceType("");
+	  this.#refreshSystemState();
+	  return;
   }
 
   public setParameter(_key: string, _value: ParameterValue): void {
     throw new Error("Parameter editing is not supported by the Rosbridge connection");
   }
 
   public publish({ topic, msg }: PublishPayload): void {
```

Comment:
can this function be implemented? if not, please leave a comment saying this is not yet implemented, however I think our rosboard transport does support sending messages from the client to the server

Reply:
We can expand our [custom rosboard client module](packages/studio-base/src/players/rosboardClient.ts) easily to achieve that.

Reply:
We are gonna left this as a future work